frontend/src/components/professor/GameManagement/AddChapterModal/AddOrEditChapterModal.js

```diff
@@ -83,24 +110,18 @@ export function AddChapterModal({ showModal, setShowModal, refetchChapterList, i
   }
 
   return (
-    images && (
+    images &&
+    currentActivityValues !== undefined && (
```

Comment:
Ja bym to zrobił inaczej. Bo zauważ, że to sprawdzenie jest zawsze true. `currentActivityValues` jest zawsze !== undefined. Dlaczego ? Bo `null` to nie `undefined`, a domyślnie `currentActivityValues == null`. Ja bym wgl zrobił tak:
```jsx
const [activityValues, setActivityValues] = useState(EMPTY_INITIAL_VALUES)
```
Potem w ifie:
```jsx
setActivityValues({
      name: name,
      sizeX: sizeX,
      sizeY: sizeY,
      posX: posX,
      posY: posY,
      imageId: imageId
})
```
Dalej wywaliłbym to sprawdzenie `currentActivityValues !== undefined` bo to zawsze true. Idąc dalej:
```jsx
initialValues={activityValues} // zamiast initialValues={currentActivityValues ? currentActivityValues : EMPTY_INITIAL_VALUES}
```
EDIT: jeżeli użyłbyś `state'a` to już bym sprawdził czy `activityValues !== undefined` jest potrzebne. Bo chyba najpierw jest `undefined`, a potem ustawia na `EMPTY_INITIAL_VALUES`

Reply:
Sprawdziłem, z tego co rozumiem nie jest potrzebne - useState ustawia value na pierwszy render